test/unit/theory/theory_int_opt_white.cpp

```diff
@@ -0,0 +1,126 @@
+#include <iostream>
+#include "smt/optimization_solver.h"
+#include "test_smt.h"
+
+namespace CVC4 {
+
+using namespace theory;
+using namespace smt;
+
+namespace test {
+
+class TestTheoryWhiteIntOpt : public TestSmt
```

Comment:
Now that I think about it, does the test go through? I feel like you would need `TestSmtNoFinishInit` here and then call `d_smtEngine->finishInit()` after setting the options (after line 21).

Reply:
Strangely the test passes, but I can definitely change it to TestSmtNoFinishInit and call finishInit after.

Reply:
I've made the change to NoFinishInit. Just as a note: the test broke when I remove the options I needed, so it seems like the options were somehow getting set correctly. Just wanted to let you know in case this behavior is unexpected.

Reply:
Hm, but "incremental" is not needed? So the while loop in `checkOpt` is not entered at all in the tests? Can you construct a test where the while loop is covered, too? You will have to reenable "incremental" for that.

Reply:
I realized that we don't need incremental for the parent solver, we just need it for the subsolver (d_optslv). Since the subsolver sets its own options we don't have to worry about setting incremental for the parent. We do need produce_assertions, however, since that allows us to push the assertions down from the parent to the subsolver.